packages/curve-ui-kit/src/shared/ui/DataTable.tsx

```diff
@@ -0,0 +1,155 @@
+import { useRef } from 'react'
+import Typography from '@mui/material/Typography'
+import TableHead from '@mui/material/TableHead'
+import TableCell from '@mui/material/TableCell'
+import TableBody from '@mui/material/TableBody'
+import Table from '@mui/material/Table'
+import TableRow from '@mui/material/TableRow'
+import { SizesAndSpaces } from '../../themes/design/1_sizes_spaces'
+import { useIntersectionObserver } from 'ui'
+import {
+  Cell,
+  Column,
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  Header,
+  Row,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table'
+import { useSortFromQueryString } from '../../hooks/useSortFromQueryString'
+import { ArrowDownIcon } from '../icons/ArrowIcon'
+
+const { Sizing, Spacing, MinWidth } = SizesAndSpaces
+
+const getAlignment = <T extends any>({ columnDef }: Column<T>) => (columnDef.meta?.type == 'numeric' ? 'right' : 'left')
+
+const DataCell = <T extends unknown>({ cell }: { cell: Cell<T, unknown> }) => {
+  const column = cell.column
+  return (
+    <Typography
+      variant="tableCellMBold"
+      color="text.primary"
```

Comment:
Not related to this PR, but should we add explicit typing here? Sometimes we use `text.primary`, sometimes `textPrimary`; either works. But there's no auto-complete so it's not consistent across the board and you have to look it up somewhere each time.

EDIT: likewise for the variant

Reply:
The types for typography seem to be a bit screwed up. Variants work fine for e.g. in buttons giving errors because of the d.ts overrides

Reply:
I've created a PR to fix typography variant typing in https://github.com/curvefi/curve-frontend/pull/557

Reply:
thanks!!